apps/src/templates/studioHomepages/Sections.jsx

```diff
@@ -58,23 +75,33 @@ const Sections = React.createClass({
             result={this.state.sectionsResult}
             nameOrId={this.state.sectionsResultName}
           />
-          {sections.length > 0 && (
+          {isTeacher && numTeacherSections > 0 && sectionFlow2017 && (
+            <SectionsPage
+              className="sectionPage"
+              teacherHomepage={this.props.teacherHomepage}
+            />
+          )}
+          {(
+            (isTeacher && numTeacherSections > 0 && !sectionFlow2017)
+            ||
+            (isStudent && numStudentSections > 0)
+          ) && (
             <SectionsTable
-              sections={sections}
+              sections={studentSections}
               isRtl={isRtl}
               isTeacher={isTeacher}
               canLeave={canLeave}
               updateSections={this.updateSections}
               updateSectionsResult={this.updateSectionsResult}
             />
           )}
-          {sections.length === 0 && isTeacher && (
-            <SectionsSetUpMessage
-              codeOrgUrlPrefix={codeOrgUrlPrefix}
-              isRtl={isRtl}
-            />
+          {isTeacher && numTeacherSections === 0 && (
+            <SectionsSetUpMessage isRtl={isRtl}/>
           )}
-          {!isTeacher && (
+          {isTeacher && numTeacherSections === 0 && sectionFlow2017 && (
+            <AddSectionDialog handleImportOpen={() => {}/* TODO */}/>
+          )}
+          {isStudent && (
```

Comment:
Yeah, but I figured that'd be a big/scary enough refactor that it should happen later.  Are you fine with me doing that in this PR?  I promise to write good tests :grin: